spec.bs

```diff
@@ -853,6 +860,27 @@ To <dfn>fetch script</dfn> given a [=URL=] |url|:
   1. Return |script|.
 </div>
 
+<div algorithm>
+To <dfn>fetch WebAssembly</dfn> given a [=URL=] |url|:
+
+  1. Let |request| be the result of [=creating a request=] with |url|, "`application/wasm`", and
+    null.
+  1. Let |moduleObject| be null.
+  1. [=Fetch=] |request| with [=fetch/processResponseConsumeBody=] set to the following steps given
+    a [=response=] |response| and |responseBody|:
+    1. Set |moduleObject| to failure and return, if any of the following conditions hold:
+      * |responseBody| is null or failure;
+      * [=header list/getting a structured field value|Getting=] "X-Allow-Protected-Audience" from
+        |response|'s [=response/header list=] does not return true.
+    1. Let |moduleObjectPromise| be the result of [=WebAssembly compileStreaming=] |response|.

```

Comment:
This should not refer to the DOM API but instead the internal algorithm that the DOM API delegates to: https://webassembly.github.io/spec/web-api/#compile-a-potential-webassembly-response

In any case, these steps are all in parallel and cannot construct and resolve/wait for resolution of Promises because there is no global here. How does this work in the implementation?

Reply:
[It calls](https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:content/services/auction_worklet/auction_v8_helper.cc;l=543;drc=857aeecd0342189fd4e0ef0ee0e1bef6b0ac2546;bpv=1;bpt=1) v8::WasmModuleObject::Compile(responseBody), with the responseBody buffer.
Thought about using [compile-a-webassembly-module](https://www.w3.org/TR/wasm-js-api/#compile-a-webassembly-module), but it's defined in the WebAssembly JavaScript Interface spec.

Reply:
@domfarolino friendly ping.

Reply:
Yep I think that is a better definition to link to!

Reply:
Just in case you're interested: one of the reasons we can't refer to DOM APIs directly in the spec is because it implies that specs call APIs that JS can override in a funky way which we don't want to allow. Instead we always want to call "internal" APIs that can't be modified directly by script. See the following

 - https://github.com/w3c/IntersectionObserver/issues/55#issue-110714442
 - (https://github.com/w3c/IntersectionObserver/issues/401#issue-525036970)
 - (https://github.com/w3c/IntersectionObserver/issues/427)

Reply:
Good to know, thanks!